### PR TITLE
{system} [intel-201{4b,5a}] eudev 3.0 (REVIEW) 

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by hajgato** [comment]
I will leave the WIP status, but in theory it is ready.
To make the binary hwdb database one have to run 
udevadm hwdb --update command, and this will create a file 
${prefix}/etc/udev/hwdb.bin

**Comment by boegel** [comment]
@hajgato: any idea when the hwdb is required exactly? any documentation on that?

**Comment by hajgato** [comment]
@boegel: http://www.linuxfromscratch.org/lfs/view/development/chapter06/eudev.html
Information about hardware devices is maintained in the /etc/udev/hwdb.d and /usr/lib/udev/hwdb.d directories. Eudev needs that information to be compiled into a binary database /etc/udev/hwdb.bin. Create the initial database:
LD_LIBRARY_PATH=/tools/lib udevadm hwdb --update
This command needs to be run each time the hardware information is updated.

**Comment by hajgato** [comment]
@boegel: there is a chance that it is only needed at boot time, but I do not know...

**Comment by hajgato** [comment]
@boegel: therie is a chance that eudev have to be compiled for each kernel (sub)version, as it communicate with it, so this project will not be an easy ride.

**Comment by boegel** [comment]
@hajgato: should we have an easyblock for `eudev` then, which injects the kernel version as a versionsuffix in the module name being installed?

**Comment by hajgato** [comment]
@boegel Probably. But I do not know whether it really needs. In theory you need the kernel-dev modules (not in the easyconfig yet). But anyway, I would firs test whether it works, and then we can do wider tests, and based on the results, modifications.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e132a748bd5e5b6e82b8 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/aa4708cdfee96eebd518 for a full test report.

File: easybuild/easyconfigs/g/gperf/gperf-3.0.4-intel-2014b.eb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+toolchain = {'name': 'intel', 'version': '2014b'}
+
+source_urls = [GNU_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+sanity_check_paths = {
+                      'files': ['bin/gperf'],
+                      'dirs': []
```
Comment (boegel) [inline easybuild/easyconfigs/g/gperf/gperf-3.0.4-intel-2014b.eb]: @hajgato: please fix indentation here?

```python
sanity_check_paths = {
    'files': ['bin/gperf'],
    'dirs': [],
}
```

File: easybuild/easyconfigs/g/gperf/gperf-3.0.4-intel-2014b.eb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+name = 'gperf'
+version = '3.0.4'
+
+homepage = 'http://www.gnu.org/software/gperf/'
+description = """GNU gperf is a perfect hash function generator. For a given list of strings, it produces a hash
+ function and hash table, in form of C or C++ code, for looking up a value depending on the input string. The hash
+ function is perfect, which means that the hash table has no collisions, and the hash table lookup needs a single
+ string comparison only."""
+
+
```
Comment (boegel) [inline easybuild/easyconfigs/g/gperf/gperf-3.0.4-intel-2014b.eb]: remove duplicate empty line

**Comment by boegel** [comment]
Just minor style issues, should be good to go in once those are tackled.

I'm already testing #1226 on top of the eudev provided here...

**Comment by hajgato** [comment]
@boegel Need some more testing. Whether it is working on different kernel what was compiled, whether we need kernel-dev for comilation (I guess Yes, but did not checked)

**Comment by hajgato** [comment]
I confirmed that no kernel-devel needed for the compilation. (on CentOS 7)

**Comment by boegel** [comment]
OK, so this should be good to go?

**Comment by wpoely86** [comment]
Are you really sure about that? Seems pretty weird to me?

**Comment by stdweird** [comment]
@hajgato fyi `systemd-udev` is now standalone https://wiki.gnome.org/Projects/libgudev

**Comment by hajgato** [comment]
@stdweird And does that provide `libudev` or only `libgudev` ?

**Comment by stdweird** [comment]
@hajgato afaik, it is libudev (not sure though)

**Comment by boegel** [comment]
No, it's not, from the README (at https://git.gnome.org/browse/libgudev/tree/README):

```
libgudev requires:
  >=libudev-199
  >=glib2-2.22.0
  pkg-config
```

**Comment by stdweird** [comment]
@hajgato or not, seems libgudev requires libudev... hmm confused now

**Comment by wpoely86** [comment]
@stdweird it's just the gobject bindings for libudev. It's the glib interface for udev. not the real low-level interface. I think libudev is way too intertangled with udev to split it off.

**Comment by boegel** [comment]
some minor style fixes I'd like to see included in https://github.com/hajgato/easybuild-easyconfigs/pull/20

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f3d1d989b29e949d9ac4 for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**